src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -0,0 +1,228 @@
+    )
+  }
+
+  def getWdl(methodConfig: MethodConfiguration): ReadWriteAction[String] = {
+    withMethod(methodConfig.methodRepoMethod.methodNamespace, methodConfig.methodRepoMethod.methodName, methodConfig.methodRepoMethod.methodVersion, getUserInfo(credential)) { method =>
+      withWdl(method) { wdl =>
+        DBIO.successful(wdl)
+      }
+    }
+  }
+
+  //submit the batch of workflows with the given ids
+  def submitWorkflowBatch(workflowIds: Seq[Long]): Future[WorkflowSubmissionMessage] = {
+
+    val workflowBatchFuture = dataSource.inTransaction { dataAccess =>
+      for {
+      //Load a bunch of things we'll need to reconstruct information:
+      //The list of workflows in this submission
+        wfRecs <- dataAccess.workflowQuery.findWorkflowByIds(workflowIds).result
+        submissionId = wfRecs.head.submissionId
+        assert(wfRecs.forall(_.submissionId == submissionId)) //sanity check they're all in the same submission
+        unpackedWfOpts <- DBIO.sequence(wfRecs.map(dataAccess.workflowQuery.loadWorkflow)) //reify to real workflows
+        assert(unpackedWfOpts.size == wfRecs.size) //there should be the right number of them
+        assert(unpackedWfOpts.forall(_.isDefined)) //and they should all have been found
+        unpackedWfs = for {wfOpt <- unpackedWfOpts; wf <- wfOpt} yield {
+          wf
+        }
+
+        //The submission itself
+        subRecs <- dataAccess.submissionQuery.findById(submissionId).result
+        submissionRec <- subRecs.headOption
+
+        //The workspace
+        wsRecs <- dataAccess.workspaceQuery.findByIdQuery(submissionRec.workspaceId).result
+        workspaceRec <- wsRecs.headOption
+
+        //The workspace's billing project
+        bpOpt <- dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspaceRec.namespace))
+        billingProject <- bpOpt
+
+        //The person who submitted the submission, and their token
+        submitterOpt <- dataAccess.rawlsUserQuery.load(RawlsUserRef(RawlsUserSubjectId(submissionRec.submitterId)))
+        submitter <- submitterOpt
+        tokenOpt <- DBIO.from(googleServicesDAO.getToken(submitter))
+        token <- tokenOpt
+
+        //The wdl
+        mcOpt <- dataAccess.methodConfigurationQuery.loadMethodConfigurationById(submissionRec.methodConfigurationId)
+        methodConfig <- mcOpt
+        wdl <- getWdl(methodConfig)
+      } yield {
+
+        val wfOpts = buildWorkflowOpts(workspaceRec, submissionId, submitter, token, billingProject)
+
+        val wfInputsBatch = unpackedWfs map { wf =>
+          val methodProps = wf.inputResolutions map {
+            case svv: SubmissionValidationValue if svv.value.isDefined =>
+              svv.inputName -> svv.value.get
+          }
+          MethodConfigResolver.propertiesToWdlInputs(methodProps.toMap)
+        }
+
+        //yield the things we're going to submit to Cromwell
+        (wdl, wfRecs.map(_.id), wfInputsBatch, wfOpts)
+      }
+    }
```

Comment:
Probably pull this out into a separate function since it's a bit of a mess. `GetMeTheStuffToSubmitToCromwell`.